packages/contracts-bedrock/deploy/001-AddressManager.ts

```diff
@@ -15,6 +19,29 @@ const deployFn: DeployFunction = async (hre) => {
       await assertContractVariable(contract, 'owner', deployer)
     },
   })
+
+  const addressManager = await getDeploymentAddress(hre, 'AddressManager')
+
+  // Transfer the address manager to the final system owner.
+  if ((await addressManager.owner()) !== hre.deployConfig.finalSystemOwner) {
```

Comment:
This logic should go in the `postDeployAction` hook, the contract is right there